### PR TITLE
Feat/client metrics page

### PR DESCRIPTION
**Pull request by Oigres2**

### CODE REVIEW

File: app/src/main/java/ipvc/gymbuddy/app/adapters/MetricAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            val date = parser.parse(dateStr)
+            if (date != null) {
+                formatter.format(date)
+            } else {
+                "Invalid date"
+            }
+        } catch (e: Exception) {
+            "Invalid date"
+        }
+    }
```
Comment (paulovareiro29) [inline app/src/main/java/ipvc/gymbuddy/app/adapters/MetricAdapter.kt]: use DateUtils object. if it doesn't exist there, create it there

File: app/src/main/java/ipvc/gymbuddy/app/datastore/MetricDataStore.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    var metrics = MutableLiveData<AsyncData<List<Metric>>>(AsyncData(listOf()))
+    var metric = MutableLiveData<AsyncData<Metric>>(AsyncData())
+
+    fun getMetrics() {
+        metrics.postValue(AsyncData(metrics.value?.data ?: listOf(), AsyncData.Status.LOADING))
+        coroutine.launch {
+            when (val response = metricService.getMetrics()) {
+                is RequestResult.Success -> {
+                    Log.d("MetricDataStore", "API Response: ${response.data}")
```
Comment (paulovareiro29) [inline app/src/main/java/ipvc/gymbuddy/app/datastore/MetricDataStore.kt]: REMOVE ALL LOGS

File: app/src/main/java/ipvc/gymbuddy/app/fragments/client/ClientHomeFragment.kt [hunk 3]
```diff
@@ -28,6 +28,8 @@ class ClientHomeFragment : BaseFragment<FragmentClientHomeBinding>(
         binding.name.text = authViewModel.user.value?.name ?: "Anonymous"
 
         loadMetrics()
+
+        binding.myMetricsButton.setOnClickListener { navController.navigate(R.id.clientMetricsOverviewFragment) }
```
Comment (paulovareiro29) [inline app/src/main/java/ipvc/gymbuddy/app/fragments/client/ClientHomeFragment.kt]:
```suggestion
        binding.myMetricsButton.setOnClickListener { navController.navigate(R.id.client_metrics_overview_fragment) }
```

File: app/src/main/java/ipvc/gymbuddy/app/viewmodels/client/ClientMetricOverViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+class ClientMetricOverViewModel(application: Application) : BaseViewModel(application) {
+    private val metricDataStore = MetricDataStore.getInstance(application)
+    val metricsData = metricDataStore.metrics
+
+    fun getMetrics() {
+        metricDataStore.getMetrics()
+        metricsData.observeForever { data ->
+            Log.d("MetricViewModel", "Data received: ${data.status} - Size: ${data.data?.size}")
+        }
```
Comment (paulovareiro29) [inline app/src/main/java/ipvc/gymbuddy/app/viewmodels/client/ClientMetricOverViewModel.kt]: remove

File: app/src/main/res/menu/client_sidebar_menu.xml [hunk 5]
```diff
@@ -9,7 +9,10 @@
     <item android:title="@string/client">
         <menu>
             <group>
-
+                <item
+                    android:id="@+id/clientMetricsOverviewFragment"
```
Comment (paulovareiro29) [inline app/src/main/res/menu/client_sidebar_menu.xml]:
```suggestion
                    android:id="@+id/sidebar_client_metrics_overview_item"
```

**Review by paulovareiro29 — CHANGES_REQUESTED** [review]

**Review by paulovareiro29 — DISMISSED** [review]
Update with changes from develop

**Review by DiogoBernardes — APPROVED** [review]

**Pull request merged**